### PR TITLE
Fix contracts page not working with lazy loading

### PR DESCRIPTION
**Pull request by MohamedElmdary**
### Description
Fix contracts page not working with lazy loading

### Changes
- fix: app was loading clients before checking is completed so we defer all components to avoid this behaviour
- fix: profile manage popup wasn't correctly loaded so we defer it's poping till it's fully loaded
image

### Related Issues
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3322

### Tested Scenarios
- check if error is happening due to lazy components for example; try to use url which is not yet set

### Documentation PR

For UI changes, Please provide the Documetation PR on [info_grid](https://github.com/threefoldtech/info_grid)

#### Preliminary Checks:

- Does it completely address the issue linked?
- What about edge cases?
- Does it meet the specified acceptance criteria?
- Are there any unintended side effects?
- Does the PR adhere to the team's coding conventions, style guides, and best practices?
- Does it integrate well with existing features?
- Does it impact the overall performance of the application?
- Are there any bottlenecks or slowdowns?
- Has it been optimized for efficiency?
- Has it been adequately tested with unit, integration, and end-to-end tests?
- Are there any known defects or issues?
- Is the code well-documented?
- Are changes to documentation reflected in the code?

#### UI Checks:

- If a UI design is provided/ does it follow it?
- Does every button work?
- Is the data displayed logical? Is it what you expected?
- Does every validation work?
- Does this interface feel intuitive?
- What about slow network? Offline?
- What if the gridproxy/graphql/chain is failing?
- What would a first time user have a hard time navigating here?

#### Code Quality Checks:

- Code formatted/linted? Are there unused imports? .. etc
- Is there redundant/repeated code?
- Are there conditionals that are always true or always false?
- Can we write this more concisely?
- Can we reuse this code? If yes, where?
- Will the changes be easy to maintain and update in the future?
- Can this code become too complex to understand for other devs?
- Can this code cause future integration problems?

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by 0oM4R** [comment]
image
the tft price not works fine looks like the issue is still exists

File: packages/playground/src/components/logger.vue [hunk 1]
```diff
@@ -203,7 +203,7 @@ export default {
     let _interceptorQueue: LI[] = [];
 
     async function interceptMessage(instance: LI) {
-      if (connectDB.value.error) {
+      if (connectDB?.value?.error) {
```
Comment (0oM4R) [inline packages/playground/src/components/logger.vue:206]: there are still some errors coming from the logger 
https://github.com/threefoldtech/tfgrid-sdk-ts/blob/431d55a9b56dff5a2ffc4011f0da6bb704a465b0/packages/playground/src/components/logger.vue#L215
image

**Comment by 0oM4R** [comment]
can you please explain the solution please?

File: packages/playground/src/components/logger.vue (hunk 1 above)
Comment (MohamedElmdary) [reply packages/playground/src/components/logger.vue:206]: this issue caused due to nullable prototype of object

image

**Comment by MohamedElmdary** [comment]
> can you please explain the solution please?

here we define promise based lock
image

which allows us to await it later untill monitor service is resolved
image

awaiting it will block resolving component which blocking loading any clients file untill we release that lock

**Review by maayarosama — APPROVED** [review]
Works fine with me 
[Screencast from 2024-09-30 11-37-39.webm](https://github.com/user-attachments/assets/8ef530a7-10b4-4c7d-ba4b-e8fd123774ba)

**Comment by 0oM4R** [comment]
tested: 
- reload on contracts page 
- reload on deployments page 
- open contracts url in incognito
all works fine with me,  only the incognito have errors in console but this issue is reported  https://github.com/threefoldtech/tfgrid-sdk-ts/issues/3342

**Review by 0oM4R — APPROVED** [review]

**Pull request merged**